lib/rubocop/cop/style/safe_navigation.rb

```diff
@@ -120,13 +120,22 @@ def autocorrect(node)
             corrector.remove(begin_range(node, body))
             corrector.remove(end_range(node, body))
             corrector.insert_before(method_call.loc.dot, '&')
+            handle_comments(corrector, method_call)
 
             add_safe_nav_to_all_methods_in_chain(corrector, method_call, body)
           end
         end
 
         private
 
+        def handle_comments(corrector, method_call)
+          return if processed_source.comments.empty?
+
+          comments = processed_source.comments.map(&:text).join("\n")
+          corrector.insert_before(method_call.loc.expression,
+                                  comments + "\n")
```

Comment:
This seems to insert all the comments from the file, not matter where they are located.

Reply:
You're absolutely right! This is something we all missed. Thanks for the heads-up.

@tejasbubane Can you take a look?

Reply:
The problem is reported in #7397 and fixed in #7420.